sylt-parser/src/parser.rs

```diff
@@ -182,43 +182,36 @@ pub struct Type {
     pub kind: TypeKind,
 }
 
-/// Tokens and their line numbers.
-type Tokens = [(T, usize)];
-
 type ParseResult<'t, T> = Result<(Context<'t>, T), (Context<'t>, Vec<Error>)>;
 
 /// Keeps track of where the parser is currently parsing.
 #[derive(Debug, Copy, Clone)]
 pub struct Context<'a> {
     pub skip_newlines: bool,
     /// All tokens to be parsed.
-    pub tokens: &'a Tokens,
-    /// The index of the curren token in the tokens slice.
+    pub tokens: &'a [Token],
+    /// The corresponding span for each token. Matches 1:1 with the tokens.
```

Comment:
Why split them out of their tuples?

Reply:
At some point I think I had to, but currently it's only because I think this:
```rust
    let (ctx, kind) = match &ctx.tokens[ctx.curr..] {
        // ...
        [T::Identifier(name), T::ColonColon, ..] => {
```
is nicer to read than this:
```rust
    let (ctx, kind) = match &ctx.tokens[ctx.curr..] {
        // ...
        [(T::Identifier(name), _), (T::ColonColon, _), ..] => {
```